roles/matrix-nginx-proxy/templates/nginx/conf.d/matrix-jitsi.conf.j2

```diff
@@ -54,6 +54,20 @@
 		tcp_nodelay on;
 	}
 
+	# colibri (JVB) websockets for additional JVBs
+	location ~ ^/colibri-ws/([a-zA-Z0-9-\.]+)/(.*) {
+		proxy_pass http://$1:9090/colibri-ws/$1/$2$is_args$args;
```

Comment:
It sounds dangerous to just get something from the URL path and to use it as a host.

It might expose various hosts that we try to keep private and don't reverse-proxy to.

I wonder if we can tighten this up somehow (regex whitelist with known allowed hosts; some prefix; etc.)?

Reply:
Yes you're right, in my scenario I'm using a private IP address here which is only accessible to nginx but in hindsight you are correct this is probably a bit of a security hole. I'll have a think.

Reply:
The problem is that if you Matrix server is public, one could hit:

- `https://jitsi.DOMAIN/colibri-ws/google.com/search` and hit `http://google.com:9090/colibri-ws/search`, etc.
- `https://jitsi.DOMAIN/colibri-ws/matrix-synapse/search` and hit `http://matrix-synapse:9090/colibri-ws/search`, etc.

Thankfully, none of these services are exposed on the `9090` port, and none offer services under the`/colibri-ws` path prefix, so it can't do too much damage.

Because of this, we may consider merging this as is.

It would have been better if it only handled whitelsited hosts though. It may be with some mapping (host id to hostname). By passing it to a `matrix_jitsi_*` variable, and then wiring that to some `matrix_nginx_proxy_..jittsi..*` variable (via `group_vars/matrix_servers`), we could pass the mapping to this file and it could generate a new block for each service.

By default, the mapping could include `jvb-1`, pointing to `matrix-jitsi-jvb:9090`.

One could add more things to it:

- `jvb-2` pointing to `https://jvb.DOMAIN`
- `jvb-3` pointing to `http://matrix-jvb-3:9090`
- etc.

The mapping keys (`jvb-1`, `jvb-2`, `jvb-3`, etc.) are probably useful for passing to Jitsi somehow? How does it know to invoke these other JVB servers otherwise? Is one meant to redefine `JVB_WS_SERVER_ID={{ matrix_jitsi_jvb_server_id }}` or change `matrix_jitsi_jvb_server_id` to a comma-separated list of JVB ids (`jvb-1,`jvb-2`)?

Would it be hard to change to something like this?